cfgov/v1/management/commands/archive_pages.py

```diff
@@ -0,0 +1,124 @@
+            ),
+        )
+        parser.add_argument(
+            "--days",
+            type=int,
+            default=0,
+            help=(
+                "Archive content published this number of days ago. "
+                "Combines with --years --months. Default: 0"
+            ),
+        )
+        parser.add_argument(
+            "--by-published-date",
+            choices=["first", "last"],
+            default="first",
+            help=(
+                "Archive based on either first or last published date. "
+                "Default: first"
+            ),
+        )
+
+    def handle(self, *args, **options):
+        url_paths = options["url_path"]
+
+        # Get the current date/time and then get our cutoff date for archiving
+        # based on it.
+        archived_at = timezone.now()
+        cutoff_date = archived_at - relativedelta(
+            years=options["years"],
+            months=options["months"],
+            days=options["days"],
+        )
+
+        # Construct a Q object to filter on based on this command-line argument
+        if options["by_published_date"] == "last":
+            published_date_filter = Q(last_published_at__lt=cutoff_date)
+        else:
+            published_date_filter = Q(first_published_at__lt=cutoff_date)
+
+        # We"ll use Wagtail"s page routing to resolve the page at the given the
+        # URL paths
+        default_site = Site.objects.get(is_default_site=True)
+        root_page = default_site.root_page
+
+        for path in url_paths:
+            path_components = path.split("/")
+
+            # Get the filterable list page we"re interested in
+            try:
+                filterable_page = root_page.route(None, path_components).page
+            except Http404:
+                raise CommandError(
+                    f"Unable to find a filterable list page for {path}. "
+                    "Ensure that the path is correct, and leave off any "
+                    "leading or trailing / characters."
+                )
+
+            # Get the filterable list queryset and filter it
+            filtered_pages = filterable_page.get_filterable_queryset().filter(
+                published_date_filter,
+                is_archived="no",
+            )
+
+            # Archive the content, letting the user know the title of the
+            # filterable list page, the cuttoff date and how many pages will be
+            # archived.
+            self.stdout.write(
+                f"Found {filtered_pages.count()} pages within "
+                f"{filterable_page.title} older than {cutoff_date}"
+            )
+            with transaction.atomic():
+                update_count = filtered_pages.select_for_update().update(
+                    is_archived="yes",
+                    archived_at=archived_at
+                )
+            self.stdout.write(
+                f"Archived {update_count} pages under {filterable_page.title}"
+            )
```

Comment:
How would you feel about combining the output into a single "Found and archived _n_ unarchived pages within <parent> older than <cutoff_date>" message?

Reply:
Done in 1e70b31.
It now appears like:

```
Found and archived 994 pages within Blog older than 2018-11-09 14:26 UTC.
```